? extend_modes.patch
Index: media.fields.inc
===================================================================
RCS file: /cvs/drupal-contrib/contributions/modules/media/media.fields.inc,v
retrieving revision 1.22
diff -u -p -r1.22 media.fields.inc
--- media.fields.inc	16 Sep 2010 23:15:39 -0000	1.22
+++ media.fields.inc	27 Oct 2010 20:07:13 -0000
@@ -90,6 +90,7 @@ function media_field_view_modes($obj_typ
       'media_original' => array('label' => t('Original')),
     );
   }
+  drupal_alter('media_field_view_modes', $modes);
   return $modes;
 }
 
