Index: wysiwyg.js
===================================================================
--- wysiwyg.js	(revision 43)
+++ wysiwyg.js	(working copy)
@@ -120,7 +120,7 @@
     }
     // Attach editor, if enabled by default or last state was enabled.
     if (params.status) {
-      Drupal.wysiwyg.editor.attach[params.editor](context, params, (Drupal.settings.wysiwyg.configs[params.editor] ? jQuery.extend(true, {}, Drupal.settings.wysiwyg.configs[params.editor][params.format]) : {}));
+      Drupal.wysiwyg.editor.attach[params.editor](context, params, (Drupal.settings.wysiwyg.configs[params.editor] ? jQuery.extend(true, {}, Drupal.settings.wysiwyg.configs[params.editor][params.format][params.field]) : {}));
     }
     // Otherwise, attach default behaviors.
     else {
Index: wysiwyg.module
===================================================================
--- wysiwyg.module	(revision 43)
+++ wysiwyg.module	(working copy)
@@ -156,10 +156,14 @@
       // Check editor theme (and reset it if not/no longer available).
       $theme = wysiwyg_get_editor_themes($profile, (isset($profile->settings['theme']) ? $profile->settings['theme'] : ''));
 
+      // Rough estimate of textarea height in pixels:
+      // 40 for the toolbar, 29 for the first row, and 15 for the rest.
+      $profile->settings['height'] = !empty($field['#rows']) ? 40 + 29 + 15 * ($field['#rows'] - 1) : 400;
+
       // Add plugin settings (first) for this text format.
       wysiwyg_add_plugin_settings($profile);
-      // Add profile settings for this text format.
-      wysiwyg_add_editor_settings($profile, $theme);
+      // Add profile settings for this field and text format.
+      wysiwyg_add_editor_settings($field, $profile, $theme);
     }
   }
   // Use a hidden element for a single text format.
@@ -345,20 +349,13 @@
 }
 
 /**
- * Add editor settings for a given input format.
+ * Add editor settings for a given field and text format.
  */
-function wysiwyg_add_editor_settings($profile, $theme) {
-  static $formats = array();
-
-  if (!isset($formats[$profile->format])) {
-    $config = wysiwyg_get_editor_config($profile, $theme);
-    // drupal_to_js() does not properly convert numeric array keys, so we need
-    // to use a string instead of the format id.
-    if ($config) {
-      drupal_add_js(array('wysiwyg' => array('configs' => array($profile->editor => array('format' . $profile->format => $config)))), 'setting');
-    }
-    $formats[$profile->format] = TRUE;
-  }
+function wysiwyg_add_editor_settings($field, $profile, $theme) {
+  $config = wysiwyg_get_editor_config($profile, $theme);
+  // drupal_to_js() does not properly convert numeric array keys, so we need
+  // to use a string instead of the format id.
+  drupal_add_js(array('wysiwyg' => array('configs' => array($profile->editor => array('format' . $profile->format => array($field['#id'] => $config))))), 'setting');
 }
 
 /**
Index: editors/ckeditor.inc
===================================================================
--- editors/ckeditor.inc	(revision 43)
+++ editors/ckeditor.inc	(working copy)
@@ -131,7 +131,7 @@
     'width' => '100%',
     // For better compatibility with smaller textareas.
     'resize_minWidth' => 450,
-    'height' => 420,
+    'height' => isset($config['height']) ? $config['height'] : 400,
     // @todo Do not use skins as themes and add separate skin handling.
     'theme' => 'default',
     'skin' => !empty($theme) ? $theme : 'kama',
@@ -326,4 +326,3 @@
   }
   return $plugins;
 }
-
