diff --git integerdate.module integerdate.module
index 0bf1336..0bf87b1 100644
--- integerdate.module
+++ integerdate.module
@@ -145,8 +145,10 @@ function integerdate_field_formatter_view($entity_type, $entity, $field, $instan
   $settings = $display['settings'];
   $element = array();
 
-  $formatted_date = integerdate_field_formatter_format_date($items[0]['value'], $settings['format_option'], $settings['format_custom_string']);
-  $element[0] = array('#markup' => $formatted_date);
+  foreach ($items as $delta => $item) {
+    $formatted_date = integerdate_field_formatter_format_date($item['value'], $settings['format_option'], $settings['format_custom_string']);
+    $element[$delta] = array('#markup' => $formatted_date);
+  }
 
   return $element;
 }
