Index: views_handler_area_link.inc
===================================================================
RCS file: /cvs/drupal-contrib/contributions/modules/views_linkarea/views_handler_area_link.inc,v
retrieving revision 1.1.2.2.2.1
diff -u -p -r1.1.2.2.2.1 views_handler_area_link.inc
--- views_handler_area_link.inc	18 Nov 2010 20:17:19 -0000	1.1.2.2.2.1
+++ views_handler_area_link.inc	13 Dec 2010 17:20:42 -0000
@@ -90,17 +90,13 @@ class views_handler_area_link extends vi
         // not work.  This should get switched to a multi-value interface of
         // some kind instead of ugly string parsing. @todo
         $querystring = strtr($this->options['querystring'], $tokens);
-        foreach (explode('&', $querystring) as $fragment) {
-          list($key, $value) = explode('=', $fragment);
-          $query[$key] = $value;
-        }
-        $link_options['query'] =$query;
+        $link_options['query'] = drupal_get_query_array($querystring);
       }
       if (!empty($this->options['anchor'])) {
         $link_options['anchor'] = strtr($this->options['anchor'], $tokens);
       }
       if (!empty($this->options['class'])) {
-        $link_options['attributes'] = array('class' => strtr($this->options['class'], $tokens));
+        $link_options['attributes'] = array('class' => explode(' ', strtr($this->options['class'], $tokens)));
       }
 
       return l(strtr($this->options['text'], $tokens), strtr($this->options['path'], $tokens), $link_options);
