Index: apachesolr_search.module
===================================================================
--- apachesolr_search.module	(revision 161)
+++ apachesolr_search.module	(working copy)
@@ -665,9 +665,14 @@
  *   The result array for this record to which to add.
  */
 function apachesolr_search_node_result($doc, &$result, &$extra) {
+  // We've pre-loaded the formatted username in the solr index, so we can just
+  // make the link manually here.  This is the only safe way to not have to do
+  // a full user_load() on every node object.  Blech.
+  $username = user_access('access user profiles') ? l($doc->name, 'user/' . $doc->uid) : $doc->name;
+
   $result += array(
     'type' => apachesolr_search_get_type($doc->type),
-    'user' => theme('username', array('account' => $doc)),
+    'user' => $username,
     'date' => $doc->created,
     'node' => $doc,
   );
