Index: field_group.module
===================================================================
--- field_group.module	(revision 118)
+++ field_group.module	(working copy)
@@ -555,6 +555,7 @@
     case 'htabs':
       $element += array(
         '#type' => 'horizontal_tabs',
+        '#title' => check_plain(t($group->label)),
         '#weight' => $group->weight,
         '#theme_wrappers' => array('horizontal_tabs'),
         '#prefix' => '<div class="field-group-' . $group->format_type . '-wrapper ' . $classes . '">',
@@ -857,7 +858,7 @@
   // Add required JavaScript and Stylesheet.
   drupal_add_library('field_group', 'horizontal-tabs');
 
-  $output = '<h2 class="element-invisible">' . t('Horizontal Tabs') . '</h2>';
+  $output = '<h2>' . $element['#title'] . '</h2>';
   $output .= '<div class="horizontal-tabs-panes">' . $element['#children'] . '</div>';
 
   return $output;
