Index: horizontal-tabs/horizontal-tabs.js
===================================================================
--- horizontal-tabs/horizontal-tabs.js	(revision 116)
+++ horizontal-tabs/horizontal-tabs.js	(working copy)
@@ -45,6 +45,13 @@
         }
       });
 
+      // Add required fields mark to any fieldsets containing required fields
+      $('fieldset.horizontal-tabs-pane').each(function(i){
+       if ($(this).find('.form-required').length > 0) {
+         $('li.horizontal-tab-button').eq(i).children('a').find('strong:first').after($('.form-required').eq(0).clone()).after('&nbsp;');
+       }
+      });
+
       $('> li:first', tab_list).addClass('first');
       $('> li:last', tab_list).addClass('last');
 
